### PR TITLE
update some libanki stuff related to notes and notetypes

### PR DESCRIPTION
**Pull request by BrayanDSO**
## Purpose / Description
prep for porting the template previewer

## Approach
In the commits

## How Has This Been Tested?

Unit tests + Did the same steps of #14692 to be sure it wasn't created again.

## Checklist
_Please, go through these checks before submitting the PR._

- [X] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by david-allison** [comment]
Unit tests failure

File: AnkiDroid/src/main/java/com/ichi2/anki/CardTemplatePreviewer.kt [hunk 1]
```diff
@@ -393,7 +392,8 @@ open class CardTemplatePreviewer : AbstractFlashcardViewer() {
             i++
         }
         try {
-            return EphemeralCard.fromNote(n, getColUnsafe, cardIndex)
+            val ord = n.cardIndexToOrd(getColUnsafe, cardIndex)
+            return n.ephemeralCard(getColUnsafe, ord = ord, customNoteType = notetype)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardTemplatePreviewer.kt:396]: Have you tested this with

* https://github.com/ankidroid/Anki-Android/issues/14692#issuecomment-1808380554

File: AnkiDroid/src/main/java/com/ichi2/anki/CardTemplatePreviewer.kt [hunk 2]
```diff
@@ -450,41 +450,6 @@ open class CardTemplatePreviewer : AbstractFlashcardViewer() {
     }
 }
 
-private class EphemeralCard(col: Collection, id: Long?) : Card(col, id) {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardTemplatePreviewer.kt:453]: This feels like it needs a separate commit

**Review by david-allison — CHANGES_REQUESTED** [review]
needs testing

**Review by david-allison — APPROVED** [review]

**Review by lukstbit — APPROVED** [review]

**Pull request merged**